tests/suites/test_suite_platform_printf.data

```diff
@@ -55,3 +55,44 @@ printf_int:"%d":-2147483648:"-2147483648"
 
 printf "%d", -0x80000000
 printf_int:"%d":-0x80000000:"-2147483648"
+
+# The next few test cases exercise how the test framework handles special
+# characters in strings.
+printf "%c", SPACE, SPACE
+printf_char2:"%c%c":SPACE_CHAR:SPACE_CHAR:"  "
```

Comment:
May I ask what's the  criteria for the cases? It seems we don't cover all the combination of "special characters".
And what's the difference between `printf "%c"` and `printf "%c%c"`? Is it a typo?

Reply:
The test cases cover all the characters that are potentially special to the test framework (newline, backslash, double quote, colon — plus space and `n` as ordinary characters). The sequences are chosen mostly to validate backslash processing. There's no strong logic to the exact set of test cases. Some of them are mostly to validate that the old weird handling of backslashes is no longer in place.

`%c` was remains of an early draft.